src/web/rustdoc.rs

```diff
@@ -290,7 +290,7 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
         );
         let url = ctry!(req, Url::parse(&redirect_path));
 
-        Ok(super::redirect(url))
+        Ok(Response::with((status::MovedPermanently, Redirect(url))))
```

Comment:
why this change? This redirect isn't used in the badge-handler, also it changes the temporary redirect to a permanent one. 

I didn't check the impact of it, but this probably should be a separate PR about redirects with doc pages?

Reply:
uhhh good point, I think this is left over from trying to get the tests to pass. Reverted now.